### PR TITLE
docs: add cert renewal availability impact

### PR DESCRIPTION
**Pull request by lennessyy**
## Describe the Change

This PR adds explanation of impact on cluster availability during the cert renewal process. 

## Changed Pages

💻 [Renew Certs](https://deploy-preview-3872--docs-spectrocloud.netlify.app/clusters/cluster-management/certificate-management/)

## Jira Tickets

🎫 [Jira Ticket]()

## Backports

Can this PR be backported?

- [ ] No. This is a release PR.

### CODE REVIEW

File: docs/docs-content/clusters/cluster-management/certificate-management.md [hunk 1]
```diff
@@ -27,6 +26,17 @@ years.
 
 :::
 
+## Certificate Renewal Impact on Cluster Availability
+
+Both automatic and manual renewal leverage the `kubeadm certs renew` command. For more information about
+`kubectl certs`, refer to
```
Comment (karl-cardenas-coding) [inline docs/docs-content/clusters/cluster-management/certificate-management.md:32]:
```suggestion
`kubectl certs` command, refer to
```
Tiny suggestion

**Review by karl-cardenas-coding — APPROVED** [review]

File: docs/docs-content/clusters/cluster-management/certificate-management.md [hunk 2]
```diff
@@ -27,6 +26,17 @@ years.
 
 :::
 
+## Certificate Renewal Impact on Cluster Availability
+
+Both automatic and manual renewal leverage the `kubeadm certs renew` command. For more information about
```
Comment (antongisli) [inline docs/docs-content/clusters/cluster-management/certificate-management.md:31]: this is not accurate. for k3s and rke2, we restart the k8s rke2-server/k3s-server. 
this does not impact workloads, but may impact cluster management availability during restart, depending on the distro.
you should state that we use different methods depending on the distro, but that the user doesn't need to do anything

File: docs/docs-content/clusters/cluster-management/certificate-management.md [hunk 3]
```diff
@@ -27,6 +26,17 @@ years.
...
+`kubectl certs`, refer to
+[Kubernetes documentation](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-certs/).
+
+After the new certificates are generated, the cluster needs to restart the control plane components in order to pick up
+the new certificates. If you have a single-node control plane, this will cause a short period of downtime for the
+control plane components, while your worker nodes will continue to operate normally in the meantime. If you have three
+nodes or more in your control plane, certificate renewal will cause zero downtime.
```
Comment (antongisli) [inline docs/docs-content/clusters/cluster-management/certificate-management.md:38]:
```suggestion
nodes or more in your control plane, certificate renewal will cause zero downtime for both workloads and control-plane elements.
```

File: docs/docs-content/clusters/cluster-management/certificate-management.md [hunk 4]
```diff
@@ -13,12 +13,11 @@ recommend you review the
 [PKI certificates and requirement](https://kubernetes.io/docs/setup/best-practices/certificates) Kubernetes
 documentation to learn more about the auto-generated certificates and to better understand their purpose.
```
Comment (antongisli) [inline docs/docs-content/clusters/cluster-management/certificate-management.md:14]: somewhere in this doc we need to explain how automatic renewals behave. 30 days before expiry, etc.

File: docs/docs-content/clusters/cluster-management/certificate-management.md [hunk 5]
```diff
@@ -13,12 +13,11 @@ recommend you review the
...
 
-This reference page focuses on how to renew the PKI certificates through Palette. You have two options for how you can
-renew the cluster PKI certificates:
+This page focuses on how to renew the PKI certificates through Palette. You have two options for how you can renew the
```
Comment (antongisli) [inline docs/docs-content/clusters/cluster-management/certificate-management.md:16]: somewhere in this doc we need to explain how automatic renewals behave. 30 days before expiry, etc.

**Review by antongisli — CHANGES_REQUESTED** [review]
There are several items that must be addressed in this document before it is good to go.
Automatic behaviours (30 days before expiry, etc), combine with a note that you can see the age/days of all certs in the listing.
Behaviours during renewal, single node, multinode control plane, impact on workloads, impact on the cluster availabilty (as distinctly different things)
even with single CP node, workloads are not affected while the CP is restarting.

File: docs/docs-content/clusters/cluster-management/certificate-management.md (hunk 5 above)
Comment (lennessyy) [reply docs/docs-content/clusters/cluster-management/certificate-management.md:16]: This is in the Automatic Certificate Renewal section

**Comment by lennessyy** [comment]
I removed references to kubeadm  - both the language in renewal and in the opening paragraph where we say we install kubernetes with it. I imagine we don't use it to install kubernetes if it's k3s or rke2? either way that information is not particularly relevant so it's removed. I didn't explicitly say that the exact mechanism we use to renew varies by kubernetes distro, since this requires no user action and simply saying that the mechanism varies doesn't really give the user any useful information so it's better to just cut it and focus on the things the user has control over,

File: docs/docs-content/clusters/cluster-management/certificate-management.md [hunk 6]
```diff
@@ -27,7 +24,17 @@ years.
...
+## Certificate Renewal Impact on Cluster Availability
+
+After the new certificates are generated, the cluster needs to restart the control plane components in order to pick up
+the new certificates. Depending on your cluster structure, this restart period may impact the cluster management
+availability of your cluster, but will not affect your cluster workloads:
+
+- If you have a single-node control plane, this will cause a short period of downtime for the control plane components,
+  meaning that cluster management availability will be temporarily impacted. However, your worker nodes will continue to
+  operate normally in the meantime and their workloads will not be impacted.
+- If you have three nodes or more in your control plane, certificate renewal will not cause downtown for neither the
```
Comment (antongisli) [inline docs/docs-content/clusters/cluster-management/certificate-management.md:36]:
```suggestion
- If you have three nodes or more in your control plane, certificate renewal will not cause downtime for neither the
```

File: docs/docs-content/clusters/cluster-management/certificate-management.md [hunk 7]
```diff
@@ -39,9 +46,8 @@ You can learn more about each option in the following sections.
 
 ## Automatic Certificate Renewal
 
-Palette will automatically update the cluster PKI certificates 30 days before they expire. The automatic renewal process
-will not cause a cluster repave. The automatic renewal process will only renew the certificates for the control plane
-nodes.
+Palette will automatically update the cluster PKI certificates 30 days before they expire. You can view the issue and
+expiry date of the cluster by click on **View K8s Certificates** in the cluster details page.
```
Comment (antongisli) [inline docs/docs-content/clusters/cluster-management/certificate-management.md:50]:
```suggestion
expiry date of the cluster by clicking on **View K8s Certificates** in the cluster details page.
```

File: docs/docs-content/clusters/edge/cluster-management/certificate-renewal.md [hunk 8]
```diff
@@ -12,7 +12,9 @@ certificates allows Kubernetes to secure API connections, verify the authenticit
 All certificates have an expiry date, and need to be renewed periodically.
 
 This page guides you through the different methods used to renew certificates in an airgapped Palette Edge cluster. An
-airgapped cluster means a cluster that has no connection to a Palette instance.
+airgapped cluster means a cluster that has no connection to a Palette instance. For information on how to renew
+certificates on connected Edge clusters, refer to
+[Renew Cluster PKI Certificates](../../cluster-management/certificate-management.md).
 
 ## Limitations
```
Comment (antongisli) [inline docs/docs-content/clusters/edge/cluster-management/certificate-renewal.md:19]: do we need to add the limitation (known issue) that auto cert renewal for connected clusters requires them to be connected? I saw there was a known issue mmentioned by eng. 
Also, what happens if the cluster reconnects at say day t-25 (and hubble tried to do the renewal at t-30), will it then do the renewal?

**Review by antongisli — APPROVED** [review]

File: docs/docs-content/clusters/edge/cluster-management/certificate-renewal.md (hunk 8 above)
Comment (lennessyy) [reply docs/docs-content/clusters/edge/cluster-management/certificate-renewal.md:19]: We usually don't add known issues to the limitations section. Limitations are more like expected behavior - they may or may not be removed in the future. But I will clarify regarding when a disconnected Edge host reconnects.

**Pull request merged**